Add language autodetect plugin

ArtikusHG opened the pull request:
## What does this PR do?

<!-- MANDATORY -->

<!-- explain the changes in your PR, algorithms, design, architecture -->

Adds a plugin which allows the search engine to autodetect the language of the search query and search in that language.

## Why is this change important?

<!-- MANDATORY -->

<!-- explain the motivation behind your PR -->

Because it improves results drastically for people who use the search engine in different languages.

## How to test this PR locally?

<!-- commands to run the tests or instructions to test the changes-->

Just enable the plugin, and do some search queries in different languages.

Commit:
[mod] new plugin: Autodetect search language

tiekoetter commented on searx/plugins/autodetect_search_language.py:
```diff
@@ -0,0 +1,14 @@
+default_on = False
+
+js_dependencies = tuple()
+css_dependencies = tuple()
```
What is the reason for these two variables with empty tuples?

ArtikusHG replied:
They were in the example in the docs, there is no reason to keep them, I assumed it was a standard to have them in all plugins

tiekoetter replied:
You can remove them. If you want to you can clean up the branch by squashing some commits together.

ArtikusHG replied:
Okay, if there's anything else you want me to do tell me, I'm on my phone atm, but I will go back to this PR tomorrow

ArtikusHG replied:
Done, hopefully everything is good now

return42 committed:
[mod] improve 'Autodetect search language' plugin

- Add documentation to the plugin
- Harmonize FastText language model with SearXNG's language model

Reosurces::

    import fasttext                                    # --> +10 MB
    fasttext.load_model(str(data_dir / 'lid.176.ftz')) # --> +4MB

Suggested-by: @dalf

- To speed up and simplify the deployment use fasttext-wheel instead of fasttext
- Building numpy on the Alpine Linux of docker-images takes ages --> install
  py3-numpy from Alpines package manager (apk)
- Alpine Linux on docker-images (musl libc) do not support fasttext-wheel (gnu
  libc) --> patch Dockerfile and build from fastetxt:

     sed -i s/fasttext-wheel/fasttext/ requirements.txt

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 reviewed (approved):
Thanks a lot for adding lang-detection to SearXNG :+1:
